orca-clouddriver/src/test/java/com/netflix/spinnaker/orca/clouddriver/tasks/monitoreddeploy/MonitoredDeployBaseTaskTest.java

```diff
@@ -168,6 +173,99 @@ void shouldReturnDefaultLogMsgWhenUnexpectedErrorHasOccurred() {
     assertThat(logMessageOnUnexpectedError).isEqualTo("<NO RESPONSE>");
   }
 
+  @Test
+  void shouldReturnEmptyHttpErrorDetailsWhenExceptionThrownWhileReadingHttpStatus() {
+
+    var converter = new JacksonConverter(objectMapper);
+    var responseBody = new HashMap<String, String>();
+
+    responseBody.put("error", "400 - Bad request, application name cannot be empty");
+
+    Response response =
+        Mockito.spy(
+            new Response(
+                "/deployment/evaluateHealth",
+                HttpStatus.BAD_REQUEST.value(),
+                HttpStatus.BAD_REQUEST.name(),
+                Collections.emptyList(),
+                new MockTypedInput(converter, responseBody)));
+
+    Mockito.when(response.getStatus()).thenThrow(IllegalArgumentException.class);
+
+    String logMessageOnHttpError = monitoredDeployBaseTask.getRetrofitLogMessage(response);
+
+    String status = "";
+    String body = "";
+    String headers = "";
+
+    assertThat(logMessageOnHttpError)
+        .isEqualTo(
+            String.format("status: %s\nheaders: %s\nresponse body: %s", status, headers, body));
+  }
+
+  @Test
+  void shouldReturnOnlyStatusWhenExceptionThrownWhileParsingHttpErrorBody() {
```

Comment:
same here, e.g.:
```
  @Test
  void returnsOnlyStatusWhenExceptionParsingHttpErrorBody() {

    Response response = mock(Response.class);
    when(response.getStatus()).thenReturn(HttpStatus.BAD_REQUEST.value()); // arbitrary
    when(response.getReason()).thenReturn("arbitrary reason");
    when(response.getBody()).thenThrow(IllegalArgumentException.class);

    String logMessageOnHttpError = monitoredDeployBaseTask.getRetrofitLogMessage(response);

    String status = String.format("%d (%s)", response.getStatus(), response.getReason());

    assertThat(logMessageOnHttpError).isEqualTo(String.format("status: %s\nheaders: \nresponse body: ", status));
  }
```

Reply:
Simplified the test case that verifies the scenario when getRetrofitLogMessage() returns empty status.